src/main/java/org/aksw/simba/lemming/metrics/metricselection/Complexity.java

```diff
@@ -0,0 +1,45 @@
+package org.aksw.simba.lemming.metrics.metricselection;
+
+/**
+ * @author DANISH AHMED on 8/24/2018
+ */
+public class Complexity {
```

Comment:
Isn't this architecture a little bit complicated? 
Wouldn't it be easier to add a new interface, e.g., `ComplexityAwareMetric` which offers a method `calculateComplexity`? This could be implemented by all triangle metrics and in the constructor of the factory class,  sets of Node- and EdgeTriangleMetric instances are created. When asked for a metric, the factory can iterate over the instances in the set and ask them for their complexity.

Note that this assumes that metric implementations are _not_ bound to a single graph and can be reused for different graphs (which should be the case, here).